packages/components/src/InputTime/hooks/__test__/useTimePredict.test.ts

```diff
@@ -0,0 +1,141 @@
+import { act, renderHook } from "@testing-library/react";
+import { htmlTimeToCivilTime } from "../../civilTimeConversions";
+import { useTimePredict } from "../useTimePredict";
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+
+describe("useTimePredict", () => {
+  it("shouldn't predict time when the value is not empty", async () => {
+    const handleChange = jest.fn();
+    const { result } = renderHook(useTimePredict, {
+      initialProps: {
+        handleChange,
+        value: htmlTimeToCivilTime("12:00"),
+      },
+    });
+
+    act(() => result.current.setTypedTime("1"));
+    await jest.runAllTimersAsync();
+
+    expect(handleChange).not.toHaveBeenCalled();
+  });
+
+  describe("12 hours", () => {
+    beforeEach(() => {
+      jest.spyOn(Intl, "DateTimeFormat").mockReturnValue({
+        // @ts-expect-error - We don't need to mock all the methods
+        resolvedOptions: () => ({ hour12: true }),
+      });
+    });
+
+    it.each`
+      currentTime | typedTime | expected
+      ${"8:24"}   | ${"1"}    | ${"10:00"}
```

Comment:
Okay so hear me out. This rocks. This is a great test.

Is there any value in providing this to the team that's asking for it, and getting them to double check the expected values? It might be a really good way to verify it's what they're asking for.

Kind of like, alright this is what we're testing for. Does this meet your needs exactly? I'm sure you're already talking to them, so even if this comment is irrelevant because you've already done this work, this kind of test is the sort of thing we can put in front of non-tech people and say "does this line up with what you want."

Either way, great test. Great work.